src/key_gen/mod.rs

```diff
@@ -36,31 +36,31 @@ use threshold_crypto::{
 use xor_name::XorName;
 
 /// A local error while handling a message, that was not caused by that message being invalid.
-#[derive(Clone, Eq, err_derive::Error, PartialEq, Debug)]
+#[derive(Clone, Eq, thiserror::Error, PartialEq, Debug)]
```

Comment:
Could we add https://doc.rust-lang.org/reference/attributes/type_system.html to prevent breaking API if we add more errors.

Reply:
ok, added and pushed